File: frontend/src/types/session.ts
// FRONTEND_AGENT | 2026-05-10 | Session type definitions
export type SessionPhase =
  | "discovery"
  | "architecture"
  | "build"
  | "quality_gate"
  | "updates"
  | "complete"
  | "failed";

export interface Session {
  id: string;
  project_id: string;
  phase: SessionPhase;
  architecture_doc: string | null;
  research_brief: string | null;
  approved_at: string | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/session.py
# BACKEND_AGENT | 2026-05-10 | Session Pydantic schemas
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from app.models.session import SessionPhase


class SessionResponse(BaseModel):
    id: str
    project_id: str
    phase: SessionPhase
    architecture_doc: str | None
    research_brief: str | None
    approved_at: datetime | None
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ApproveArchitectureRequest(BaseModel):
    approved: bool
    feedback: str | None = Field(default=None, max_length=5000)


class UpdateRequest(BaseModel):
    change_request: str = Field(min_length=10, max_length=5000)


class ConfirmChangeRequest(BaseModel):
    approved: bool
